### ISSUE TITLE
Redefine Layers in Pre-Trained Model

### ISSUE THREAD

**Issue by EoinKenny** [keras-team/keras#11863]
When I download a pre-trained model such as VGG16, I would like to redefine the last layer

Dense (with softmax included) 

to...

Two separate layers => Dense layer and SoftMax activation layer

The reason is because I would like to use the DeepLIFT and DeepExplain libraries and they demand the model is in this format.

Is it possible to do this?

**Comment by Golbstein** [comment]
```
from keras.layers import Dense, Activation
from keras.applications.vgg16 import VGG16

vgg = VGG16(include_top=True, weights=None)
x = Dense(1000)(vgg.layers[-2].output)
x = Activation('softmax')(x)
vgg_new = Model(vgg.input, x)
```

**Comment by EoinKenny** [comment]
Thanks for the reply, but I would like to keep the original ImageNet weights?

Do you need to use set_weights() and get_weights()?

**Comment by Golbstein** [comment]
You can load the weights to your new model as well. After loading the vgg model do: model.save_weights('blabla.h5') and then vgg_new.load_weights..

**Comment by EoinKenny** [comment]
Sorry, I'm a bit confused. Let me show your the code I'm using.

`first_half = Model(inputs = model.input, outputs = model.layers[-2].output)
weights = model.layers[-1].get_weights()

model.layers.pop()
x = model.layers[-1].output
x = Dense(1000)(x)
x = Activation("softmax")(x)
final_model = Model(input=model.input, output=x)

final_model.layers[-2].set_weights(weights)`

Is this correct? Or do I need to change it?

**Comment by Golbstein** [comment]
```
from keras.layers import Dense, Activation
from keras.applications.vgg16 import VGG16
vgg = VGG16(include_top=True, weights='imagenet')
vgg.save_weights('imagenet.h5')
x = Dense(1000)(vgg.layers[-2].output)
x = Activation('softmax')(x)
vgg_new = Model(vgg.input, x)
vgg_new.load_weights('imagenet.h5')
```

**Comment by EoinKenny** [comment]
Thanks a lot.
